fiftyone/utils/complex_yolo/zoo.py

```diff
@@ -7,34 +7,65 @@
 """
 import logging
 import os
-from pkg_resources import packaging
-import warnings
+
+import numpy as np
 
 import eta.core.web as etaw
-import eta.core.utils as etau
 
 import fiftyone as fo
+import fiftyone.core.media as fom
 import fiftyone.core.utils as fou
 import fiftyone.utils.torch as fout
 import fiftyone.zoo.models as fozm
 
 fou.ensure_torch()
-import math
-import numpy as np
-import open3d as o3d
 import torch
+from torch.autograd import Variable
 
-device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
-Tensor = (
-    torch.cuda.FloatTensor if torch.cuda.is_available() else torch.FloatTensor
-)
+o3d = fou.lazy_import("open3d", callback=lambda: fou.ensure_import("open3d"))
 
-from .model import build_model
+from .model import ComplexYOLOv3
 
 
 logger = logging.getLogger(__name__)
 
 
+def apply_model(
+    samples,
+    model,
+    feature_map_field,
+    feature_group_slice=None,
```

Comment:
There were some hard-coded slice/field names in the previous implementation. When adding a generic utility that is intended to work on any user data, it is better to require the user to provide all relevant parameter names, since they could have picked anything.

There is an existing convention that `label_field="predictions"` is used by default in other parts of the code, which is why I've left it here.